workshop1/slides/src/slides.md

```diff
@@ -0,0 +1,664 @@
+6) Conclusion
+
+# The Haskell Development Environment
+
+## Setting up your environment
+
+- Go to [the Github repo](https://github.com/relex/haskell-workshop) for the workshop
+(https://github.com/relex/haskell-workshop)
+
+- Clone it locally
+
+- Go to workshop1/Exercise.md
+
+- Complete "Set up tooling" section
+
+## Stack
+
+* One of Haskell's build tools.
+* To build a project: `stack build`
+* To access the REPL: `stack repl`
+
+![](images/basic-repl.png)
+
+# Hello Syntax
+
+## Playing with the REPL
+
+* Useful tool to try out parts of your program
+* Extensively used in the workshop
+
+<!-- Live REPL Coding
+* Let binding
+* Numbers & mathematical operators
+* Strings
+* Defining a function
+-->
+
+## Hello World
+
+* Open src/Main.hs
+* Run in the REPL by typing `:main`
+
+```haskell
+main :: IO ()
+main = putStrLn "Hello World!"
+```
+
+* `main` is special:
+  - Present in every Haskell program
+  - No arguments
+  - Performs an IO action
+
+* Change the text and run it in the REPL
+
+## Functions and Types
+
+* The primary way of defining computation is with **functions**
+* **Types** describe the inputs and outputs of functions
+* Types are **enforced** by the compiler
+* Working with two languages at once :)
+
+```haskell
+identity :: Int -> Int
+identity x = x
+
+hello :: String -> String
+hello name = "Hello, " ++ name
+
+addInt :: Int -> Int -> Int
+addInt x y = x + y
+```
+
+Add these ^ to a file and try them out!
+
+## A First Function
+
+* A function that converts (some) integers to words
+* Pattern matching
+
+```haskell
+intToWord :: Int -> String
+intToWord 1 = "one"
+intToWord 2 = "two"
+intToWord 3 = "three"
+intToWord _ = "dunno"
+```
+
+* Try and write `wordToInt :: String -> Int`
+
+## Polymorphic Types
+
+* Some functions have the same behaviour for values of different types
+  * E.g. the `identity` function
+* Can generalise functions by using *type variables*
+
+```haskell
+identity :: a -> a
+identity x = x
+```
+
+* Can also *constrain* the type variables
+
+```haskell
+add :: Num a => a -> a -> a
+add x y = x + y
+```
+
+* These constraints can be defined using *type classes*
+
+## Composition
+
+* Using the output of one function as the input of another function: *"chaining functions together"*
+* Use the dot operator for composition
+
+```haskell
+f :: Int -> String
+f x = intToWord x
+
+g :: String -> Int
+g x = wordToInt x
+
+h :: Int -> Int
+h = g . f
+
+(.) :: (b -> c) -> (a -> b) -> a -> c
+(.) f g x = g (f x)
+```
+
+
+# The Data Model
+
+## Goal
+
+* Create a data structure that accurately models the data
+* Introduce Maybe
+* Introduce Lists
+
+### Approach
+
+* Define a model for a single row of data (an average candidate)
+* Apply this model to all the rows
+
+![](images/data-model.pdf)
+
+## A Single Candidate
+
+![](images/small-data.png)
+
+* A candidate has 4 attributes:
+  - constituency
+  - party
+  - sex
+  - average age
+
+* Need a way of grouping these together
+
+## Data Types
```

Comment:
Maybe have the data types section before the `A single candidate`? I think it would flow better to first see how to define types and then be given instructions on what kind of type we want them to write

Reply:
Good idea. I'll repeat the slide